sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -42,7 +47,18 @@ public void grow(int neededSize, UnsafeRow row) {
     final int length = totalSize() + neededSize;
     if (buffer.length < length) {
       // This will not happen frequently, because the buffer is re-used.
-      final byte[] tmp = new byte[length * 2];
+      final byte[] tmp;
+      try {
+        tmp = new byte[length * 2];
+      } catch (NegativeArraySizeException e) {
+        String errorMessage =
+          "NegativeArraySizeException is triggered. The current length is " + buffer.length +
+          ". The new length is " + length + " * 2. totalSize is " + totalSize() +
```

Comment:
Done. https://github.com/apache/spark/pull/10120/files#diff-c654dda59109239bce5498cbc30ab251L196 will dump the unsafe row. https://github.com/apache/spark/pull/10120/files#diff-244a70a91841bbddf6fae17c14c18ce4L378 will dump the row that is human readable.